crates/runtime/src/lib.rs

```diff
@@ -279,8 +281,8 @@ impl_runtime_apis! {
             Starknet::estimate_fee(transactions, &simulation_flags)
         }
 
-        fn re_execute_transactions(transactions_before: Vec<Transaction>, transactions_to_trace: Vec<Transaction>, with_state_diff: bool) -> Result<Result<Vec<(TransactionExecutionInfo, Option<CommitmentStateDiff>)>, SimulationError>, InternalSubstrateError> {
-            Starknet::re_execute_transactions(transactions_before, transactions_to_trace, with_state_diff)
+        fn re_execute_transactions(transactions_before: Vec<Transaction>, transactions_to_trace: Vec<Transaction>, with_state_diff: bool, with_accumulated_state_diff: bool) -> Result<ReExecutionResult, InternalSubstrateError> {
```

Comment:
This became a weird fn signature.

Let's split it into two runtimeAPIs:
```rust
fn re_execute_transactions(transactions_before: Vec<Transaction>, transactions_to_trace: Vec<Transaction>, with_state_diff: bool) -> Result<Result<Vec<(TransactionExecutionInfo, Option<CommitmentStateDiff>)>, SimulationError>, InternalSubstrateError>;

fn get_transaction_re_execution_state_diff(transactions_before: Vec<Transaction>, transactions: Vec<Transaction>) -> Result<Result<CommitmentStateDiff, SimulationError>, InternalSubstrateError>
```

Reply:
leave `re_execute_transactions` impl untouch.
But for `get_transaction_re_execution_state_diff` only call get `to_state_diff` at the very end.
This way you don't return the `TransactionExecutionInfo`, you don't waste computation